app/src/main/java/com/demo/androidfundamentals/extensions.kt

```diff
@@ -41,8 +41,8 @@ fun List<MovieModel>.getDistinctMovieYears(): List<String> {
     val movieDates: MutableList<String> = mutableListOf()
 
 
-    forEach {
-        movieDates.add(AppUtils.getMovieYear(it))
+    map {
```

Comment:
this is not collections map(). Refer this https://stackoverflow.com/questions/44738319/how-to-convert-transform-a-collection-to-another-collection-by-elements-propert